CMakeLists.txt

```diff
@@ -81,6 +81,7 @@ find_package(SndFile 1.0.29 REQUIRED)
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     find_package(ALSA REQUIRED)
 endif()
+find_package(cJSON REQUIRED)
```

Comment:
[cJSON](https://github.com/DaveGamble/cJSON) is distributed in two files, so we can add it to the project as miniz and libspng. There are many JSON libraries available, also we can write the parser ourselves as we don't need advanced functionality.

What do you think?

Reply:
I always, generally prefer to use system libraries. Else we'll have to maintain our own little zoo of pet projects.

Reply:
However, I'd prefer to keep this dependency optional, and if it's not found, there won't be animated intermission screens. :shrug:

Reply:
> However, I'd prefer to keep this dependency optional, and if it's not found, there won't be animated intermission screens. 🤷

Then we will have a lot of `#ifdef HAVE_JSON` macros everywhere. There will also be sky and possibly other id24 JSON lumps. I would prefer to use git submodules or write our JSON parser.

Reply:
Ah, okay. However, why duplicate code? The cjson library is already available on all relevant systems. We could keep an "emergency copy" if it's not found by CMake, though. 

Or do you actually *want* to write a JSON parser yourself? :wink:

Reply:
Yes, I think cJSON is quite popular.

> Or do you actually _want_ to write a JSON parser yourself? 😉

Not really 😄

Reply:
> Yes, I think cJSON is quite popular.

http://os4depot.net/?function=showfile&file=development/library/misc/json-c.lha

Reply:
> > Yes, I think cJSON is quite popular.
> 
> http://os4depot.net/?function=showfile&file=development/library/misc/json-c.lha

That's the wrong one! 😆 It looks like there is no port for AmigaOS. That should be easy, cjson is C89 with no dependencies.

Reply:
Alright, I think cjson isn't asked too much for as an additional dependency.